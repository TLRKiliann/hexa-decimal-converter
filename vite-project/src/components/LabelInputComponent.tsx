import React from 'react'

type InputLblProps = {
    value: string | undefined;
    nameValue: string;
    converterTarget: (event: React.ChangeEvent<HTMLInputElement>) => void;
    children: string | JSX.Element | JSX.Element[];
}

export default function LabelInputComponent({value, nameValue, converterTarget, children}: InputLblProps) {
    return (
        <>
            <label>{children} :</label>
            <input
                data-testid="testid-input"
                type="text"
                value={value}
                onChange={(event)=>converterTarget(event)}
                placeholder={nameValue}
            />
        </>
    )
}
