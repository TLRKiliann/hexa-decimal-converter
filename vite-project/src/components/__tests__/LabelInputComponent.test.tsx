import { test, expect } from "vitest"
import { create } from "react-test-renderer"
import LabelInputComponent from '../LabelInputComponent';
import { ChangeEvent } from "react";

test('Snapshat of LabelInputComponent', () => {
    const tree = create(<LabelInputComponent value={undefined} nameValue={""} 
        converterTarget={function (_event: ChangeEvent<HTMLInputElement>): void {
        throw new Error("Function not implemented.");
    } } children={""} />)
    expect(tree.toJSON()).toMatchSnapshot()
});