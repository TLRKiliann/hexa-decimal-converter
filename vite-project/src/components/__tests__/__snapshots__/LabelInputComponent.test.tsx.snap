// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshat of LabelInputComponent 1`] = `
[
  <label>
     :
  </label>,
  <input
    onChange={[Function]}
    placeholder=""
    type="text"
  />,
]
`;
